lib/purescript-ast/src/Language/PureScript/Types.hs

```diff
@@ -585,6 +587,17 @@ unapplyConstraints = go []
   go acc (ConstrainedType _ con ty) = go (con : acc) ty
   go acc ty = (reverse acc, ty)
 
+saturate :: S.Set Text -> SourceType -> Int -> SourceType
+saturate bound ty n = fst . runSupply 0 $ do
+  tvs <- replicateM n freshTyVar
+  return $ foldl' srcTypeApp ty (srcTypeVar <$> tvs)
+  where
+  freshTyVar = do
+    tv <- ("t" <>) . T.pack . show <$> fresh
+    if tv `elem` bound
+      then freshTyVar
+      else return tv
```

Comment:
I tried to generated sensible type variables names but perhaps `freshName` would be enough?